src/dc_receive_imf.rs

```diff
@@ -57,34 +58,51 @@ pub async fn dc_receive_imf(
     server_uid: u32,
     seen: bool,
 ) -> Result<()> {
-    dc_receive_imf_inner(context, imf_raw, server_folder, server_uid, seen, false).await
+    dc_receive_imf_inner(
+        context,
+        imf_raw,
+        server_folder,
+        server_uid,
+        seen,
+        None,
+        false,
+    )
+    .await
 }
 
+/// If `is_partial_download` is set, it contains the full message size in bytes.
+/// Do not confuse that with `replace_partial_download` that will be set when the full message is loaded later.
 pub(crate) async fn dc_receive_imf_inner(
     context: &Context,
     imf_raw: &[u8],
     server_folder: &str,
     server_uid: u32,
     seen: bool,
+    is_partial_download: Option<u32>,
     fetching_existing_messages: bool,
 ) -> Result<()> {
     info!(
         context,
-        "Receiving message {}/{}, seen={}...", server_folder, server_uid, seen
+        "Receiving message {}/{}, seen={}, partial={:?}...",
```

Comment:
Could you only output `,partial={:?}` when it's not `None`? Otherwise logs will be full of `, partial=None`, which is the case for almost every message.

Reply:
i think i remove it completely from here and add it to mimeparser.rs, there we also have the information about availability.